src/frontend/src/btc/schedulers/btc-wallet.scheduler.ts

```diff
@@ -46,100 +65,159 @@ export class BtcWalletScheduler implements Scheduler<PostMessageDataRequestBtc>
-	}: {
-		identity: OptionIdentity;
-		bitcoinNetwork: BitcoinNetwork;
-	}): Promise<CertifiedData<bigint>> {
-		const balance = await getBtcBalance({
-			identity,
-			network: bitcoinNetwork,
-			minConfirmations: BTC_BALANCE_MIN_CONFIRMATIONS
-		});
-		const certifiedBalance = {
-			data: balance,
+		bitcoinNetwork,
+		btcAddress,
+		minterCanisterId,
+		certified = true
+	}: Omit<LoadBtcWalletParams, 'shouldFetchTransactions'>): Promise<
+		CertifiedData<bigint> | undefined
+	> => {
+		// Query BTC balance only if minterCanisterId is available
+		if (!certified) {
+			if (nonNullish(minterCanisterId) && isNullish(this.store.balance)) {
+				return {
+					data: await getBalanceQuery({
+						identity,
+						network: bitcoinNetwork,
+						address: btcAddress,
+						bitcoinCanisterId: BITCOIN_CANISTER_IDS[minterCanisterId]
+					}),
+					certified: false
+				};
+			}
+
+			return;
+		}
+
+		return {
+			data: await getBtcBalance({
+				identity,
+				network: mapToSignerBitcoinNetwork({
+					network: bitcoinNetwork
+				}),
+				minConfirmations: BTC_BALANCE_MIN_CONFIRMATIONS
+			}),
 			certified: true
 		};
+	};
 
-		this.store = {
-			...this.store,
-			balance: certifiedBalance
-		};
+	private loadWalletData = async ({
+		certified,
+		identity,
+		bitcoinNetwork,
+		btcAddress,
+		minterCanisterId,
+		shouldFetchTransactions
+	}: LoadBtcWalletParams) => {
+		const balance = await this.loadBtcBalance({
+			identity,
+			bitcoinNetwork,
+			certified,
+			btcAddress,
+			minterCanisterId
+		});
 
-		return certifiedBalance;
-	}
+		// TODO: investigate and implement "update" call for BTC transactions
+		const transactions =
+			shouldFetchTransactions && !certified
+				? await this.loadBtcTransactionsData({ btcAddress })
+				: [];
+
+		return { balance, transactions };
+	};
 
-	/* TODO: The following steps need to be done:
-	 * 1. [Required] Fetch uncertified transactions via BTC transaction API.
-	 * 2. [Improvement] Query uncertified balance in oder to improve UX (signer.getBtcBalance takes ~5s to complete).
-	 * 3. [Required] Fetch certified transactions via BE endpoint (to be discussed).
-	 * */
 	private syncWallet = async ({ identity, data }: SchedulerJobData<PostMessageDataRequestBtc>) => {
 		const bitcoinNetwork = data?.bitcoinNetwork;
-		assertNonNullish(bitcoinNetwork, 'No BTC network provided to get BTC certified balance.');
+		assertNonNullish(bitcoinNetwork, 'No BTC network provided to get BTC balance.');
 
 		const btcAddress = data?.btcAddress.data;
 		assertNonNullish(btcAddress, 'No BTC address provided to get BTC transactions.');
 
-		const balance = await this.loadBtcBalance({ identity, bitcoinNetwork });
-		const { newTransactions, latestBitcoinBlockHeight } =
-			nonNullish(data) && data.shouldFetchTransactions
-				? await this.loadBtcTransactionsData({ btcAddress })
-				: { newTransactions: [], latestBitcoinBlockHeight: undefined };
+		// TODO: implement "onCertifiedError" to handle errors in update calls
```

Comment:
You implement this in next PR right? i.e. feature should not go live without handling error as we reset the UI stores that way.

Reply:
Yes, gonna be the next step.